runtime/drivers/duckdb/olap.go

```diff
@@ -630,6 +630,106 @@ func (c *connection) execWithLimits(parentCtx context.Context, stmt *drivers.Sta
 	return err
 }
 
+// convertToEnum converts a varchar col in table to an enum type.
+// Generally to be used for low cardinality varchar columns although not enforced here.
+func (c *connection) convertToEnum(ctx context.Context, table, col string) error {
+	if !c.config.ExtTableStorage {
+		return fmt.Errorf("`cast_to_enum` is only supported when `external_table_storage` is enabled")
+	}
+	c.logger.Info("convert column to enum", zap.String("table", table), zap.String("col", col))
+
+	version, exist, err := c.tableVersion(table)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return fmt.Errorf("table %q does not exist", table)
+	}
+
+	dbName := dbName(table, version)
+	enum := fmt.Sprintf("%s_enum", col)
+
+	var switchErr error
+	err = c.WithConnection(ctx, 100, true, false, func(ctx, ensuredCtx context.Context, _ *dbsql.Conn) error {
+		// check that atleast one non nil value exists in the column
+		res, err := c.Execute(ctx, &drivers.Statement{Query: fmt.Sprintf("SELECT (SELECT count(%s) FROM %s.default WHERE %s IS NOT NULL) > 0 AS cnt", safeSQLName(col), safeSQLName(dbName), safeSQLName(col))})
+		if err != nil {
+			return err
+		}
+
+		var exists bool
+		if res.Next() {
+			if err := res.Scan(&exists); err != nil {
+				_ = res.Close()
+				return err
+			}
+		}
+		_ = res.Close()
+		if !exists {
+			return fmt.Errorf("column %q can't be converted to enum, has zero non null values", col)
+		}
+
+		// scan current db and current schema
+		res, err = c.Execute(ctx, &drivers.Statement{Query: "SELECT current_database(), current_schema()"})
+		if err != nil {
+			return err
+		}
+
+		var currentDB, currentSchema string
+		if res.Next() {
+			if err := res.Scan(&currentDB, &currentSchema); err != nil {
+				_ = res.Close()
+				return err
+			}
+		}
+		_ = res.Close()
+
+		// switch to source db
+		// this is only required since duckdb has bugs around db scoped custom types
+		// TODO: remove this when https://github.com/duckdb/duckdb/pull/9622 is released
+		err = c.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("USE %s", dbName)})
+		if err != nil {
+			return fmt.Errorf("failed switch db %q: %w", dbName, err)
+		}
+		defer func() {
+			// switch to original db, notice `db.schema` just doing USE db switches context to `main` schema in the current db if doing `USE main`
+			// we want to switch to original db and schema
+			switchErr = c.Exec(ensuredCtx, &drivers.Statement{Query: fmt.Sprintf("USE %s.%s", safeSQLName(currentDB), safeSQLName(currentSchema))})
+		}()
+
+		err = c.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("DROP TYPE %s", safeSQLName(enum))})
```

Comment:
NIT: Use DROP TYPE IF EXISTS. 
Also, since we are creating a new versioned DB on every type, would we ever hit the case when enum already exists?

Reply:
You can also use CREATE OR REPLACE TYPE in the create command to avoid running DROP TYPE command.

Reply:
`CREATE OR REPLACE` doesn't exist for `TYPE` so need to drop first and then recreate. 

> Also, since we are creating a new versioned DB on every type, would we ever hit the case when enum already exists?

Yeah I was thinking there may be conflicts during refresh but I think even during refresh we create new db. Will check this and remove it if required.